regimes/sg/tax_identity.go

```diff
@@ -0,0 +1,56 @@
+package sg
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/invopop/gobl/cbc"
+	"github.com/invopop/gobl/tax"
+	"github.com/invopop/validation"
+)
+
+// Reference: https://lookuptax.com/docs/tax-identification-number/singapore-tax-id-guide#nric-number
+// Reference: https://mytax.iras.gov.sg/ESVWeb/default.aspx?target=GSTListingSearch
+// All these tax codes are valid options for a GST registration number.
+
+var (
+	taxCodeRegexps = []*regexp.Regexp{
+		regexp.MustCompile(`^(19[0-9]{2}|20[0-9]{2})\d{5}[A-Z]$`), // UEN (ROC)
+		regexp.MustCompile(`^\d{9}[A-Z]$`),                        // UEN (ROB)
+		regexp.MustCompile(`^[TS]\d{2}[A-Z]\w{1}\d{4}[A-Z]$`),     // UEN (Others)
+		regexp.MustCompile(`^[STFGM]\d{7}[A-Z]$`),                 // NIRC/FIN
+		regexp.MustCompile(`^\w{2}\d{7}[A-Z]$`),                   // GST
```

Comment:
With so many to choose from, I get the impression that maybe some companies choose to use multiple numbers at the same time on their invoices, suggesting that only one should be used for the main tax identity.

It could be that the GST makes the most sense in the tax identity, and everything else should be moved to the `org.Identity`. An additional (not obvious) option here is to switch to using the `identities` array instead of the `tax.Identity`.